### PR TITLE
[DOC] Improve documentation of `signal` module

### PR DESCRIPTION
**Pull request by NicolasGensollen**
Similarly to #3004 for `masking`, this PR proposes to make some improvements to the documentation of the `signal` module.
It also adds the function `nilearn.signal.butterworth` to the public API as this function was public but not included in the docs.

### CODE REVIEW

File: nilearn/signal.py [hunk 1]
```diff
@@ -248,44 +278,41 @@ def _check_wn(btype, freq, nyq):
...
         If False, `signals` is modified inplace, and memory consumption is
-        lower than for copy=True, though computation time is higher.
+        lower than for ``copy=True``, though computation time is higher.
 
     Returns
     -------
-    filtered_signals: numpy.ndarray
-        Signals filtered according to the parameters
+    filtered_signals : :class:`numpy.ndarray`
+        Signals filtered according to the parameters.
```
Comment (bthirion) [inline nilearn/signal.py:315]:
```suggestion
        Signals filtered according to the given parameters.
```

**Review by bthirion — COMMENTED** [review]
Looks great, thx !

**Review by bthirion — APPROVED** [review]
LGTM.

**Comment by NicolasGensollen** [comment]
Thanks, merging...

**Pull request merged**